ambry-utils/src/test/java/com.github.ambry.utils/SimpleByteBufferPoolTest.java

```diff
@@ -130,13 +148,18 @@ public void testFirstBlockedThenSucceed()
       used[i] = new CountDownLatch(1);
       consumers[i] = new BufferConsumer(size, maxBlockTimeInMs, pool, allocated[i], used[i]);
       consumerThreads[i] = new Thread(consumers[i]);
-      consumerThreads[i].start();
     }
-    allocated[0].await();
+    consumerThreads[0].start();
+    if (!allocated[0].await(100, TimeUnit.MILLISECONDS)) {
+      fail("SimpleByteBufferPool takes too long time to allocate a buffer to BufferConsumer[0].");
+    }
     assertEquals(0, pool.availableMemory());
+    consumerThreads[1].start();
```

Comment:
This was verified by indirect factors: 1) the buffer pool does not have enough memory to allocate to consumer-2, AND 2) consumer-2 has no error when it completes, which means it gets memory eventually. So between consumer-2 requesting and getting memory, the state is waiting.

To make the checking more clear, I have added an explicit check for the state of consumer-2.